app/scripts/modules/core/src/managed/ManagedResourceObject.tsx

```diff
@@ -21,23 +22,50 @@ const kindIconMap: { [kind: string]: IconNames } = {
   'application-load-balancer': 'loadBalancer',
 };
 
-function getIconTypeFromKind(kind: string) {
-  return kindIconMap[getKindName(kind)] ?? 'placeholder';
-}
+const getIconTypeFromKind = (kind: string) => kindIconMap[getKindName(kind)] ?? 'placeholder';
+
+const getResourceName = ({ moniker: { app, stack, detail } }: IManagedResourceSummary) =>
+  [app, stack, detail].filter(Boolean).join('-');
 
-export const ManagedResourceObject = ({
-  resource: {
+const getResourceRoutingInfo = (
+  resource: IManagedResourceSummary,
+): { state: string; params: { [key: string]: string } } | null => {
+  const {
     kind,
-    moniker: { app, stack, detail },
-  },
-  artifact,
-  depth,
-}: IManagedResourceObjectProps) => {
+    locations: { account },
+  } = resource;
+  const resourceType = getKindName(kind);
+  const params = {
+    acct: account,
+    q: getResourceName(resource),
+  };
+
+  switch (resourceType) {
+    case 'cluster':
+      return { state: 'home.applications.application.insight.clusters', params };
+
+    case 'security-group':
+      return { state: 'home.applications.application.insight.firewalls', params };
+
+    case 'classic-load-balancer':
+    case 'application-load-balancer':
+      return { state: 'home.applications.application.insight.loadBalancers', params };
+  }
+
+  return null;
+};
+
+export const ManagedResourceObject = ({ resource, artifact, depth }: IManagedResourceObjectProps) => {
   const { version: currentVersion, buildNumber: currentBuild } = parseName(artifact?.versions.current || '') || {};
+  const { kind } = resource;
+  const resourceName = getResourceName(resource);
+  const routingInfo = getResourceRoutingInfo(resource);
+  const route = useSref(routingInfo.state, routingInfo.params);
```

Comment:
is deref-ing `routingInfo` here going to NPE in the case where `getResourceRoutingInfo` returns `null`?

Reply:
Oops! yes. Thanks for the catch. Fixing it

Reply:
I wish `strictNullChecks` was enabled to catch these :)